proposals/geolocation.md

```diff
@@ -0,0 +1,404 @@
+# WoT Discovery
+## Geolocation
+Geolocation support for discovery includes two loosely-coupled
+specifications: an information model for geolocation data itself,
+which might be encoded in TDs or returned as data from properties,
+and a query mechanism to filter such data by location and time.
+
+Note: filtering by time of last update potentially could be broken out
+as a separate proposal since it is applicable to data other than
+location.  However, location is special since the Directory needs to
+track it to provide geolocation queries.  Queries over other kinds of
+data would require additional requirements in the Directory e.g. a 
+time-series database, and this is currently out of scope.
```

Comment:
If the directory keeps a history of changes to each TD, it could be used to track static location updates.

Reply:
True, although I think general history opens up a huge can of problems.  So I think we want to target some very specific use cases here.  Basically, if the location is "stale" then I might want to ignore it or give it a higher uncertainty, or estimate the current location based on the last known position and speed.  So the time of last update is important.  Maintaining an entire history might be useful for more accurate trajectory estimates, but I think this is a relatively minor benefit for a lot of additional requirements, privacy issues around data retention, and so forth.  To discuss...